### DISCUSSION TITLE
Making PyMuPDF Dependency Optional for Advanced Usage

### DISCUSSION THREAD

**Discussion by comfuture** [Filimoa/open-parse#54] [General]
Is it possible to make the PyMuPDF dependency optional? It appears to be used for parsing tables in the code, but I cannot find any direct imports. Since it's a mandatory dependency, installing open-parse on distributions without a wheel binary (linux/arm64) results in an incomplete compilation.

Reference: https://github.com/pymupdf/PyMuPDF/issues/3622

Would it be feasible to change the setup so that PyMuPDF is only explicitly listed as a dependency for advanced usage? This could potentially alleviate installation issues on platforms lacking precompiled binaries for PyMuPDF.

**Comment by Filimoa** [comment]
PyMuPDF is currently needed to convert pdf pages to images - this is used for the table parsing + bbox visualization.  I'd love to migrate away from pymupdf but haven't had the time. For now making it optional would cause issues, open to PR's.
